pkg/gb/instructions.go

```diff
@@ -69,1185 +65,5 @@ func (gb *Gameboy) popPC16() uint16 {
-			gb.thisCpuTicks += 4
-		}
-
-	// JP Z,nn
-	case 0xCA:
-		next := gb.popPC16()
-		if gb.CPU.Z() {
-			gb.instJump(next)
-			gb.thisCpuTicks += 4
-		}
-
-	// JP NC,nn
-	case 0xD2:
-		next := gb.popPC16()
-		if !gb.CPU.C() {
-			gb.instJump(next)
-			gb.thisCpuTicks += 4
-		}
-
-	// JP C,nn
-	case 0xDA:
-		next := gb.popPC16()
-		if gb.CPU.C() {
-			gb.instJump(next)
-			gb.thisCpuTicks += 4
-		}
-
-	// JP HL
-	case 0xE9:
-		gb.instJump(gb.CPU.HL.HiLo())
-
-	// JR n
-	case 0x18:
-		addr := int32(gb.CPU.PC) + int32(int8(gb.popPC()))
-		gb.instJump(uint16(addr))
-
-	// JR NZ,n
-	case 0x20:
-		next := int8(gb.popPC())
-		if !gb.CPU.Z() {
-			addr := int32(gb.CPU.PC) + int32(next)
-			gb.instJump(uint16(addr))
-			gb.thisCpuTicks += 4
-		}
-
-	// JR Z,n
-	case 0x28:
-		next := int8(gb.popPC())
-		if gb.CPU.Z() {
-			addr := int32(gb.CPU.PC) + int32(next)
-			gb.instJump(uint16(addr))
-			gb.thisCpuTicks += 4
-		}
-
-	// JR NC,n
-	case 0x30:
-		next := int8(gb.popPC())
-		if !gb.CPU.C() {
-			addr := int32(gb.CPU.PC) + int32(next)
-			gb.instJump(uint16(addr))
-			gb.thisCpuTicks += 4
-		}
-
-	// JR C,n
-	case 0x38:
-		next := int8(gb.popPC())
-		if gb.CPU.C() {
-			addr := int32(gb.CPU.PC) + int32(next)
-			gb.instJump(uint16(addr))
-			gb.thisCpuTicks += 4
-		}
-
-	// CALL nn
-	case 0xCD:
-		gb.instCall(gb.popPC16())
-
-	// CALL NZ,nn
-	case 0xC4:
-		next := gb.popPC16()
-		if !gb.CPU.Z() {
-			gb.instCall(next)
-			gb.thisCpuTicks += 12
-		}
-
-	// CALL Z,nn
-	case 0xCC:
-		next := gb.popPC16()
-		if gb.CPU.Z() {
-			gb.instCall(next)
-			gb.thisCpuTicks += 12
-		}
-
-	// CALL NC,nn
-	case 0xD4:
-		next := gb.popPC16()
-		if !gb.CPU.C() {
-			gb.instCall(next)
-			gb.thisCpuTicks += 12
-		}
-
-	// CALL C,nn
-	case 0xDC:
-		next := gb.popPC16()
-		if gb.CPU.C() {
-			gb.instCall(next)
-			gb.thisCpuTicks += 12
-		}
-
-	// RST 0x00
-	case 0xC7:
-		gb.instCall(0x0000)
-
-	// RST 0x08
-	case 0xCF:
-		gb.instCall(0x0008)
-
-	// RST 0x10
-	case 0xD7:
-		gb.instCall(0x0010)
-
-	// RST 0x18
-	case 0xDF:
-		gb.instCall(0x0018)
-
-	// RST 0x20
-	case 0xE7:
-		gb.instCall(0x0020)
-
-	// RST 0x28
-	case 0xEF:
-		gb.instCall(0x0028)
-
-	// RST 0x30
-	case 0xF7:
-		gb.instCall(0x0030)
-
-	// RST 0x38
-	case 0xFF:
-		gb.instCall(0x0038)
-
-	// RET
-	case 0xC9:
-		gb.instRet()
-
-	// RET NZ
-	case 0xC0:
-		if !gb.CPU.Z() {
-			gb.instRet()
-			gb.thisCpuTicks += 12
-		}
-
-	// RET Z
-	case 0xC8:
-		if gb.CPU.Z() {
-			gb.instRet()
-			gb.thisCpuTicks += 12
-		}
-
-	// RET NC
-	case 0xD0:
-		if !gb.CPU.C() {
-			gb.instRet()
-			gb.thisCpuTicks += 12
-		}
-
-	// RET C
-	case 0xD8:
-		if gb.CPU.C() {
-			gb.instRet()
-			gb.thisCpuTicks += 12
-		}
-
-	// RETI
-	case 0xD9:
-		gb.instRet()
-		gb.interruptsEnabling = true
-
-	// CB
-	case 0xCB:
-		nextInst := gb.popPC()
-		gb.thisCpuTicks += CBOpcodeCycles[nextInst] * 4
-		gb.cbInst[nextInst]()
-
-	default:
-		log.Printf("Unimplemented opcode: %#2x", opcode)
-		WaitForInput()
-	}
+	gb.mainInst[opcode]()
```

Comment:
I would prefer is the contents of `mainInstructions` were in this file rather than a separate one. Also it looks like `ExecuteOpcode` is only called in one place, so it may be better just to replace that line with `gb.mainInst[opcode]()` and remove this function.